webrecorder/browsertrix
# Crawl config detail view & edit workflow UI updates

Update crawl config view & edit pages to match new create workflow.
- Resolves https://github.com/webrecorder/browsertrix-cloud/issues/397
- Resolves https://github.com/webrecorder/browsertrix-cloud/issues/399
- Resolves https://github.com/webrecorder/browsertrix-cloud/issues/420

### Manual testing
1. Run app `yarn start`
2. Navigate to crawl config list view. Verify that clicking 3 dots menu shows "Edit crawl config" and clicking on item takes you to the edit form.
3. Interact with form. Verify you're able to tab through all steps.
4. Click "Save Changes". Verify that edits save as expected and you're taken to the detail view.
5. Click "Edit Crawl Config" and make additional edits. Verify edits save as expected.
6. Interact with "Config Versions" arrows. Verify that view updates as expected.
7. Go to a crawl detail page. Click "Config" under "Files". Verify crawl config displays as expected.

### Screenshots
**Crawl Config Detail view**:
<img width="1137" alt="Screen Shot 2022-12-19 at 11 32 24 AM" src="https://user-images.githubusercontent.com/4672952/208505315-24df65e4-98b0-4ac8-b1dd-203a657313ef.png">

**Crawl Config Detail view with last crawl**:
<img width="1143" alt="Screen Shot 2022-12-19 at 2 33 54 PM" src="https://user-images.githubusercontent.com/4672952/208539842-a626b201-6964-4eb8-bd39-6b2dce42002d.png">

**Crawl Detail action menu if it has an active config**:
<img width="186" alt="Screen Shot 2022-12-20 at 11 47 48 AM" src="https://user-images.githubusercontent.com/4672952/208754171-f25ca5aa-0175-47b3-9f3d-9f1aae9e0943.png">

**Crawl Config Edit view**:
<img width="1142" alt="Screen Shot 2022-12-20 at 3 27 57 PM" src="https://user-images.githubusercontent.com/4672952/208785575-e2cfecc4-2998-4fc6-a051-a658b1890d66.png">


**Crawl Detail Config view**:
<img width="1130" alt="Screen Shot 2022-12-19 at 2 14 54 PM" src="https://user-images.githubusercontent.com/4672952/208537187-7ba8a72c-2ed5-4c1e-9ebe-35b3f48311c5.png">
